### PR TITLE
Refactored the Personal Allowance Spec.

### PR DESCRIPTION
**Pull request by DavidWoodward37**

### CODE REVIEW

File: app/forms/nonresident/PersonalAllowanceForm.scala [hunk 1]
```diff
@@ -25,7 +25,7 @@ import uk.gov.hmrc.play.views.helpers.MoneyPounds
 
 object PersonalAllowanceForm {
 
-  def personalAllowanceForm (maxPA: BigDecimal = BigDecimal(0)) = Form (
+  def personalAllowanceForm (maxPA: BigDecimal = BigDecimal(0)): Form[PersonalAllowanceModel] = Form (
     mapping(
       "personalAllowance" -> bigDecimal
```
Comment (jameshforster) [inline app/forms/nonresident/PersonalAllowanceForm.scala]: Can you update this to a text with appropriate mandatory tests and transformer please.

File: test/controllers/CalculationControllerTests/PersonalAllowanceSpec.scala [hunk 2]
```diff
@@ -16,31 +16,26 @@
...
 import org.jsoup._
 import org.scalatest.mock.MockitoSugar
 
 import scala.concurrent.Future
 import controllers.nonresident.{PersonalAllowanceController, routes}
 import models.nonresident.PersonalAllowanceModel
-import play.api.mvc.Result
 
-class PersonalAllowanceSpec extends UnitSpec with WithFakeApplication with MockitoSugar {
+class PersonalAllowanceSpec extends UnitSpec with WithFakeApplication with MockitoSugar with FakeRequestHelper {
```
Comment (jameshforster) [inline test/controllers/CalculationControllerTests/PersonalAllowanceSpec.scala]: Please update the name to PersonalAllowanceActionSpec

File: test/forms/nonResident/PersonalAllowanceFormSpec.scala [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+      "return an invalid form with one error" in {
+        form.errors.size shouldBe 1
+      }
+
+      s"return an error message of " +
+        s"'${messages.PersonalAllowance.errorMaxLimit}${MoneyPounds(11000, 0).quantity} ${messages.PersonalAllowance.errorMaxLimitEnd}" in {
+        form.error("personalAllowance").get.message shouldBe
+          s"${messages.PersonalAllowance.errorMaxLimit}${MoneyPounds(11000, 0).quantity} ${messages.PersonalAllowance.errorMaxLimitEnd}"
+      }
+    }
```
Comment (jameshforster) [inline test/forms/nonResident/PersonalAllowanceFormSpec.scala]: Could you please add tests for the scenarios where the empty and non-numeric entries are attempted.
